napi-inl.h

```diff
@@ -2811,8 +2811,95 @@ inline void Promise::CheckCast(napi_env env, napi_value value) {
   NAPI_CHECK(result, "Promise::CheckCast", "value is not promise");
 }
 
+inline Promise::Promise() : Object() {}
+
 inline Promise::Promise(napi_env env, napi_value value) : Object(env, value) {}
 
+inline MaybeOrValue<Promise> Promise::Then(napi_value onFulfilled) const {
+  EscapableHandleScope scope(_env);
+#ifdef NODE_ADDON_API_ENABLE_MAYBE
+  Value thenMethod;
+  if (!Get("then").UnwrapTo(&thenMethod)) {
```

Comment:
Open question: using `Get("then")` returns the `then` property from this object. Do we need to concern ourselves with the possibility of the `then` being shadowed, and not being the original `then` from the `Promise`? If so, we should use `Promise.prototype.then` ... similar to how we are suggested (eg. [eslint rule](https://eslint.org/docs/latest/rules/no-prototype-builtins)) to use `Object.prototype.hasOwnProperty.call(foo, bar)` instead of `foo.hasOwnProperty(bar)`.

On the flipside, the Promise spec does specifically state how `then` should behave on a Promise object (even if it's shadowed): https://promisesaplus.com/#the-then-method

I think `Get("then")` is okay, just wanted to bring this up as a point of discussion.

Reply:
This is a very valid point, given what you brought up, I'm also okay to keep it that way in light of how `then` should behave, even if it was shadowed.